### ISSUE TITLE
axios should not use JSON.parse to parse the response when the responseType is set to 'text'

### ISSUE THREAD

**Issue by feiniao0308** [axios/axios#5719]
### Describe the bug

When the response is a string, axios parses responses using JSON.parse as the default transformResponse (https://github.com/axios/axios/blob/16b5718954d88fbefe17f0b91101d742b63209c7/lib/defaults.js#L57-L65). JSON.parse tries to make the value a Number (https://github.com/axios/axios/issues/2919), e.g. JSON.parse('646686880670135092987843') => 6.466868806701352e+23. This will cause unexpected result from the client side. 

When client set responseType is to 'text', the axios should not use JSON.parse. Instead, it should return the string directly.

**Comment by KoustavFrost** [comment]
hello @feiniao0308 I want to work on this issue, can you assign it to me? Thank you.

**Comment by feiniao0308** [comment]
> hello @feiniao0308 I want to work on this issue, can you assign it to me? Thank you.

Hi @KoustavFrost - thanks for working on it. Looks like I don't have permission to assign it. Could you tell me how I can assign it to you?

**Comment by stefanosporiazis** [comment]
I believe this is resolved.

https://github.com/axios/axios/blob/a48a63ad823fc20e5a6a705f05f09842ca49f48c/dist/axios.js#L1433-L1452

`JSON.parse` is used only when `data` is a non empty string, and the responseType is 'json' or forcedJSONParsing is true.

Can someone confirm?

**Comment by ashrth** [comment]
If JSON.parse is converting string to a number, can we handle the number instead of trying to keep it a string? 
We can use BigInt. 

`typeof data == "bigint";`

**Comment by jasonsaayman** [comment]
I dont think this is something we will be changing right now but you may pass your own function for this behaviour
